test/e2e/yurthub/yurthub.go

```diff
@@ -16,21 +16,138 @@ limitations under the License.
+	if !ycfg.YurtE2eCfg.EnableYurtAutonomy {
+		klog.Infof("not enable yurt node autonomy test, so yurt-node-autonomy will not run, and will return.")
+		return
+	}
 	var _ = framework.KubeDescribe(YURTHUB_NAMESPACE_NAME, func() {
+		var (
+			c   clientset.Interface
+			err error
+		)
 		defer ginkgo.GinkgoRecover()
+		c, err = framework.LoadClientset()
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "fail to get client set")
+
+		err = ns.DeleteNameSpace(c, YURTHUB_NAMESPACE_NAME)
+		framework.ExpectNoError(err)
+
+		ginkgo.By("yurthub create namespace")
+		_, err = ns.CreateNameSpace(c, YURTHUB_NAMESPACE_NAME)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "fail to create namespaces")
+
+		ginkgo.AfterEach(func() {
+			ginkgo.By("delete namespace:" + YURTHUB_NAMESPACE_NAME)
+			err = ns.DeleteNameSpace(c, YURTHUB_NAMESPACE_NAME)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "fail to delete created namespaces")
+		})
+
+		framework.KubeDescribe("node autonomy", func() {
+			ginkgo.It("yurt node autonomy", func() {
+				NodeManager, err := nd.NewNodeController(ycfg.YurtE2eCfg.NodeType, ycfg.YurtE2eCfg.RegionId, ycfg.YurtE2eCfg.AccessKeyId, ycfg.YurtE2eCfg.AccessKeySecret)
+				framework.ExpectNoError(err)
+
+				podName := "test-yurthub-busybox"
+				objectMeta := metav1.ObjectMeta{}
+				objectMeta.Name = podName
+				objectMeta.Namespace = YURTHUB_NAMESPACE_NAME
+				objectMeta.Labels = map[string]string{"name": podName}
+				spec := apiv1.PodSpec{}
+				container := apiv1.Container{}
+				spec.HostNetwork = true
+				spec.NodeSelector = map[string]string{YURT_NODE_LABLE: "true"}
+				container.Name = "busybox"
+				container.Image = "busybox"
+				container.Command = []string{"sleep", "3600"}
+				spec.Containers = []apiv1.Container{container}
+
+				ginkgo.By("create pod:" + podName)
+				pod, err := p.CreatePod(c, YURTHUB_NAMESPACE_NAME, objectMeta, spec)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred(), "fail create busybox deployment")
+
+				ginkgo.By("waiting running pod:" + podName)
+				err = p.VerifyPodsRunning(c, YURTHUB_NAMESPACE_NAME, podName, false, 1)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred(), "fail wait pod running")
+				klog.Infof("check pod status running Successful")
+
+				ginkgo.By("get pod info")
+				pod, err = p.GetPod(c, YURTHUB_NAMESPACE_NAME, pod.Name)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred(), "fail get pod status:"+podName)
+
+				ginkgo.By("set node " + pod.Spec.NodeName + " autonomy")
+				patchNode := apiv1.Node{
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: map[string]string{YURT_NODE_ANNOTATION: "true"},
+					},
+				}
+				patchData, err := json.Marshal(patchNode)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred(), "fail marshal patch node")
+
+				node, err := c.CoreV1().Nodes().Patch(pod.Spec.NodeName, types.StrategicMergePatchType, patchData)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred(), "fail patch node autonomy")
+
+				ginkgo.By("stop node")
```

Comment:
I think this is the place where user needs to manually restart the node if the test runs against minikube. Please add sufficient instruction in the log so that the user knows what to do next.